### PR TITLE
refactor(dashboard/user): profile view

### PR DESCRIPTION
**Pull request by CodeNamedRobin**
# Description
Refactors the entire profile view. Makes the profile view a lot neater, where no more cards are used for simple settings, instead all settings are in one user settings card.
## Related issues/external references
Many of #273
## Types of changes

- New feature _(non-breaking change which adds functionality)_
- Style _(Change that do not affect the functionality of the code)_

### CODE REVIEW

File: apps/dashboard/src/modules/user/views/UserProfileView.vue [hunk 1]
```diff
@@ -1,51 +1,31 @@
...
-        <ChangePassword />
-      </div>
-      <div class="col-6 md:col-6">
-        <ChangeApiKey v-if="isAdmin"/>
-        <div class="flex flex-row mb-2 align-items-center">
-          <h3 class="mr-3">{{ t('profile.extensiveDataAnalysis') }}</h3>
-          <InputSwitch v-model="dataAnalysis" @update:modelValue="handleChange" />
-        </div>
-      </div>
+    <div class="flex md:flex-row flex-column-reverse justify-content-between gap-8">
```
Comment (SuperVK) [inline apps/dashboard/src/modules/user/views/UserProfileView.vue]: I think gap-4 (or similiar) looks a lot better :)

**Review by SuperVK — COMMENTED** [review]
It would be nice if the buttons for pop up change to a cursor mouse on hover, and I think you might as well change the password changing to be the same as the pin change? Not sure tbh, but this feels weird while they kinda do the same

And do we want to implement Roy's suggestion while we are at it: #268

**Comment by SuperVK** [comment]
Also I can't comment on the code, but the profile view somehow fetches all users and uses that to get your GEWIS id. But this endpoint will soon become privileged so it might be nice to do this differently (besides it already being very cursed that we fetch _all_ users to just use one)

File: apps/dashboard/src/modules/user/views/UserProfileView.vue (hunk 1 above)
Comment (CodeNamedRobin) [reply apps/dashboard/src/modules/user/views/UserProfileView.vue]: Has been implemented!

**Comment by CodeNamedRobin** [comment]
> Also I can't comment on the code, but the profile view somehow fetches all users and uses that to get your GEWIS id. But this endpoint will soon become privileged so it might be nice to do this differently (besides it already being very cursed that we fetch _all_ users to just use one)

A new get individual gewis user fetch function has been implemented in the user store now.

**Review by SuperVK — APPROVED** [review]

**Pull request merged**